### ISSUE TITLE
Spyder fails when connecting to an existing kernel

### ISSUE THREAD

**Issue by KoichiFunaya** [spyder-ide/spyder#18836]
### What steps will reproduce the problem?

I tried to connect remote Spyder kernel to the Spyder console at local PC. Here are the steps I took.

I run the command below at a remote server.

python -m spyder_kernels.console - matplotlib='inline' --ip=127.0.0.1 -f=./remotemachine.json

I made sure all the ports in the remotemachine.json are tunneled to the local machine.

I also copied the remotemachine.json to the local machine.

Then at the Spyder console in the local PC, I selected Consoles --> Connect to an existing kernel.

In the dialog box, I selected the remotemachine.json file I copied in the local machine and pushed "OK".

Then the Spyder immediately reports an error. The error message is as below.

At both the remote and the local machines, the Spyder and the Spyder-Kernel versions are exactly the same.

spyder : 5.1.5
spyder-kernels : 2.1.3

Any idea how I can make it run?

---

Traceback (most recent call last):
  File "C:\Users\funaya\Anaconda3\envs\e3nn\lib\site-packages\spyder\plugins\ipythonconsole\plugin.py", line 1105, in create_client_for_kernel
    self._create_client_for_kernel(connection_file, hostname, sshkey,
  File "C:\Users\funaya\Anaconda3\envs\e3nn\lib\site-packages\spyder\plugins\ipythonconsole\plugin.py", line 1935, in _create_client_for_kernel
    shellwidget.check_spyder_kernel()
  File "C:\Users\funaya\Anaconda3\envs\e3nn\lib\site-packages\spyder\plugins\ipythonconsole\widgets\shell.py", line 266, in check_spyder_kernel
    self.silent_exec_method(code)
  File "C:\Users\funaya\Anaconda3\envs\e3nn\lib\site-packages\spyder\plugins\ipythonconsole\widgets\shell.py", line 667, in silent_exec_method
    self._request_info['execute'][msg_id] = self._ExecutionRequest(msg_id,
TypeError: <lambda>() missing 1 required positional argument: 'hidden'

### Traceback
```python-traceback
Traceback (most recent call last):
  File "C:\Users\funaya\Anaconda3\envs\e3nn\lib\site-packages\spyder\plugins\ipythonconsole\plugin.py", line 1105, in create_client_for_kernel
    self._create_client_for_kernel(connection_file, hostname, sshkey,
  File "C:\Users\funaya\Anaconda3\envs\e3nn\lib\site-packages\spyder\plugins\ipythonconsole\plugin.py", line 1935, in _create_client_for_kernel
    shellwidget.check_spyder_kernel()
  File "C:\Users\funaya\Anaconda3\envs\e3nn\lib\site-packages\spyder\plugins\ipythonconsole\widgets\shell.py", line 266, in check_spyder_kernel
    self.silent_exec_method(code)
  File "C:\Users\funaya\Anaconda3\envs\e3nn\lib\site-packages\spyder\plugins\ipythonconsole\widgets\shell.py", line 667, in silent_exec_method
    self._request_info['execute'][msg_id] = self._ExecutionRequest(msg_id,
TypeError: <lambda>() missing 1 required positional argument: 'hidden'
```

## Versions

* Spyder version: 5.1.5 
* Python version: 3.9.12
* Qt version: 5.9.7
* PyQt5 version: 5.9.2
* Operating System: Windows 10

### Dependencies

```

# Mandatory:
atomicwrites >=1.2.0          :  1.4.0 (OK)
chardet >=2.0.0               :  4.0.0 (OK)
cloudpickle >=0.5.0           :  2.0.0 (OK)
cookiecutter >=1.6.0          :  1.7.3 (OK)
diff_match_patch >=20181111   :  20200713 (OK)
intervaltree >=3.0.2          :  3.1.0 (OK)
IPython >=7.6.0               :  8.4.0 (OK)
jedi >=0.17.2;<0.19.0         :  0.18.1 (OK)
jsonschema >=3.2.0            :  4.4.0 (OK)
keyring >=17.0.0              :  23.4.0 (OK)
nbconvert >=4.0               :  6.4.4 (OK)
numpydoc >=0.6.0              :  1.4.0 (OK)
paramiko >=2.4.0              :  2.8.1 (OK)
parso >=0.7.0;<0.9.0          :  0.8.3 (OK)
pexpect >=4.4.0               :  4.8.0 (OK)
pickleshare >=0.4             :  0.7.5 (OK)
psutil >=5.3                  :  5.9.0 (OK)
pygments >=2.0                :  2.11.2 (OK)
pylint >=2.5.0;<2.10.0        :  2.9.6 (OK)
pyls_spyder >=0.4.0           :  0.4.0 (OK)
pylsp >=1.2.2;<1.3.0          :  1.2.4 (OK)
pylsp_black >=1.0.0           :  None (OK)
qdarkstyle =3.0.2             :  3.0.2 (OK)
qstylizer >=0.1.10            :  0.1.10 (OK)
qtawesome >=1.0.2             :  1.0.3 (OK)
qtconsole >=5.1.0             :  5.3.1 (OK)
qtpy >=1.5.0                  :  2.0.1 (OK)
rtree >=0.9.7                 :  0.9.7 (OK)
setuptools >=49.6.0           :  61.2.0 (OK)
sphinx >=0.6.6                :  5.0.2 (OK)
spyder_kernels >=2.1.1;<2.2.0 :  2.1.3 (OK)
textdistance >=4.2.0          :  4.2.1 (OK)
three_merge >=0.1.1           :  0.1.1 (OK)
watchdog >=0.10.3             :  2.1.6 (OK)
zmq >=17                      :  23.2.0 (OK)

# Optional:
cython >=0.21                 :  None (OK)
matplotlib >=2.0.0            :  None (OK)
numpy >=1.7                   :  None (OK)
pandas >=1.1.1                :  None (OK)
scipy >=0.17.0                :  None (OK)
sympy >=0.7.3                 :  None (OK)
```

**Comment by KoichiFunaya** [comment]
Thanks. I will try and see if it works as you suggested.

**Comment by KoichiFunaya** [comment]
Yes, it worked, as mentioned at https://github.com/spyder-ide/spyder/issues/17772. Thank you!

I wonder why it didn't work in the previous setting, though. I did "conda update spyder" on both remote and local machine, but it still didn't work.

**Comment by dalthviz** [comment]
Glad it worked for you @KoichiFunaya ! The thing is that the default anaconda channel goes just up to the outdated version 5.1.5 so you need to use the conda-forge channel to get the latest version available (currently 5.3.2). Basically doing `conda update spyder` will say that you have Spyder up to date since you have Spyder 5.1.5, which is the latest release available in the default anaconda channel. There is not much we can do from the Spyder side to help with that besides recommending using the conda-forge channel or our standalone installers :/

**Comment by KoichiFunaya** [comment]
Thank you for the clarification. I now understood clearly. Of course I have to search for the latest updates at conda-forge.

Thanks again!
